### PR TITLE
stat/distuv: Extend test coverage for Bernoulli distribution and fix some functions

### PR DESCRIPTION
**Pull request by romanwerpachowski**
This change modifies Gonum to provide almost 100% test coverage for `Bernoulli` distribution type in `stats/distuv`, fix `Quantile` and `Entropy` functions and other minor fixes.

Please take a look.

### CODE REVIEW

**Comment by romanwerpachowski** [comment]
AppVeyor build failed with what looks like a transient error:
```
Build started
git clone -q https://github.com/gonum/gonum.git c:\gopath\src\gonum.org\v1\gonum
git fetch -q origin +refs/pull/1337/merge:
git checkout -qf FETCH_HEAD
Restoring build cache
Cache 'C:\Users\appveyor\AppData\Local\go-build' - Restored
%GOROOT%\bin\go version
go version go1.13.3 windows/amd64
%GOROOT%\bin\go get -v -t ./...
go: golang.org/x/exp@v0.0.0-20190125153040-c74c464bbbf2: Get https://proxy.golang.org/golang.org/x/exp/@v/v0.0.0-20190125153040-c74c464bbbf2.mod: dial tcp: lookup proxy.golang.org: no such host
Command exited with code 1
```

**Comment by kortschak** [comment]
I kicked it.

File: stat/distuv/bernoulli.go [hunk 1]
```diff
@@ -83,15 +80,22 @@ func (Bernoulli) NumParameters() int {
...
+		return 1 - b.P
+	}
+	if x == 1 {
+		return b.P
+	}
+	return 0.
 }
 
-// Quantile returns the inverse of the cumulative probability distribution.
+// Quantile returns the minimum value of x from amongst all those values whose CDF value exceeds or equals p.
```
Comment (kortschak) [inline stat/distuv/bernoulli.go:92]: Please revert this.

File: stat/distuv/bernoulli.go [hunk 2]
```diff
@@ -83,15 +80,22 @@ func (Bernoulli) NumParameters() int {
...
 func (b Bernoulli) Quantile(p float64) float64 {
 	if p < 0 || 1 < p {
 		panic(badPercentile)
 	}
-	if p < 1-b.P {
+	// Q(p) := min {x | p <= CDF(x) }
```
Comment (kortschak) [inline stat/distuv/bernoulli.go:97]: We don't need this comment, it says exactly the same thing as the code.

File: stat/distuv/bernoulli_test.go [hunk 3]
```diff
@@ -5,38 +5,161 @@
...
 	"testing"
 
 	"golang.org/x/exp/rand"
 )
 
-func TestBernouilli(t *testing.T) {
+func TestBernoulli(t *testing.T) {
 	t.Parallel()
 	src := rand.New(rand.NewSource(1))
-	for i, dist := range []Bernoulli{
```
Comment (kortschak) [inline stat/distuv/bernoulli_test.go:17]: Please leave this as `dist`.

File: stat/distuv/bernoulli_test.go [hunk 4]
```diff
@@ -5,38 +5,161 @@
...
+			}
+			if p != 0 {
+				t.Errorf("Probability for x = %g is not 0, got: %g", x, p)
+			}
+		}
 	}
 }
 
-func testBernouilli(t *testing.T, dist Bernoulli, i int) {
+func testBernoulli(t *testing.T, b Bernoulli, i int) {
```
Comment (kortschak) [inline stat/distuv/bernoulli_test.go:51]: Same here.

File: stat/distuv/bernoulli_test.go [hunk 5]
```diff
@@ -5,38 +5,161 @@
...
+		}
+	}
+	if b.P != 0.5 {
+		checkMedian(t, i, x, b, tol)
+	} else if b.Median() != 0.5 {
+		t.Errorf("Median for P == 0.5 is not 0.5")
+	}
+}
+
+func testBernoulliCDF(t *testing.T, b Bernoulli) {
```
Comment (kortschak) [inline stat/distuv/bernoulli_test.go:91]: s/b/dist/g

File: stat/distuv/bernoulli_test.go [hunk 6]
```diff
@@ -5,38 +5,161 @@
...
+	}
+	if b.CDF(1) != 1 {
+		t.Errorf("Bernoulli CDF at one is not one")
+	}
+	if b.CDF(1.00001) != 1 {
+		t.Errorf("Bernoulli CDF above one is not one")
+	}
+}
+
+func testBernoulliSurvival(t *testing.T, b Bernoulli) {
```
Comment (kortschak) [inline stat/distuv/bernoulli_test.go:112]: s/b/dist/g

File: stat/distuv/bernoulli_test.go [hunk 7]
```diff
@@ -5,38 +5,161 @@
...
+	}
+	if b.Survival(1) != 0 {
+		t.Errorf("Bernoulli Survival at one is not zero")
+	}
+	if b.Survival(1.00001) != 0 {
+		t.Errorf("Bernoulli Survival above one is not zero")
+	}
+}
+
+func testBernoulliQuantile(t *testing.T, b Bernoulli) {
```
Comment (kortschak) [inline stat/distuv/bernoulli_test.go:130]: s/b/dist/g

File: stat/distuv/bernoulli.go (hunk 1 above)
Comment (romanwerpachowski) [reply stat/distuv/bernoulli.go:92]: I can, but can I ask why? The CDF of a Bernoulli distribution does not really have an inverse. The new comment is more precise mathematically.

File: stat/distuv/bernoulli.go (hunk 2 above)
Comment (romanwerpachowski) [reply stat/distuv/bernoulli.go:97]: In a more explicit way... but OK.

File: stat/distuv/bernoulli.go (hunk 1 above)
Comment (kortschak) [reply stat/distuv/bernoulli.go:92]: Yeah, OK.

**Comment by romanwerpachowski** [comment]
PTAL.

**Comment by kortschak** [comment]
Happy 1337 PR!

**Pull request merged**